src/convertToObject.js

```diff
@@ -12,7 +12,16 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
-}
+  // eslint-disable-next-line max-len
+  const arr = sourceString.replace(/\s+|\n/g, ' ')
+    .split(';').join(':').split(':')
+    .filter(el => el.length > 1);
```

Comment:
you can replace this filter method with checking if elements exist before adding them to an object